src/cairoWriter.ts

```diff
@@ -851,7 +852,8 @@ class IdentifierWriter extends CairoASTNodeWriter {
           FunctionVisibility.External &&
         node.getClosestParentByType(IndexAccess) === undefined &&
         node.getClosestParentByType(MemberAccess) === undefined) ||
-        (node.parent instanceof FunctionCall && isExternalCall(node.parent)))
+        (node.parent instanceof FunctionCall && isExternalCall(node.parent)) ||
+        (node.parent instanceof FunctionCall && isHashFunction(node.parent)))
```

Comment:
CairoFunctionDefinitions have an acceptsRawDArray property, that can be set during creation with createCairoFunctionStub. I think that should be used rather than this

Reply:
The problem with that is, that it generates variables that don't exist. Assume `str` is the calldata string as the input to function, `str_len` and `str_ptr` becomes its parameters. Then it combines these into `str_dstruct` as a struct. Then, when the acceptsRawDArray property is set to true, its parameters become `str_dstruct_len, str_dstruct_ptr` which basically does not exist.

Reply:
cleared the cairowriter and moved conditions to utils